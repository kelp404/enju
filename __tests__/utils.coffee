config = require 'config'
elasticsearch = require 'elasticsearch'
utils = require '../lib/utils'


jest.mock 'elasticsearch'

beforeEach ->
    elasticsearch.Client.mockClear()

test 'Get elasticsearch client.', ->
    elasticsearch.Client = jest.fn (args) ->
        {
            host
            apiVersion
        } = args
    expect(utils.getElasticsearch()).toMatchSnapshot()
    expect(elasticsearch.Client).toBeCalled()

test 'Get index prefix.', ->
    expect(utils.getIndexPrefix()).toBe config.enju.indexPrefix
