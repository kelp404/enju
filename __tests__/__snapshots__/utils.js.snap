// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bleach regex words. 1`] = `"\\\\^\\\\$\\\\*\\\\+\\\\?\\\\{\\\\}\\\\.\\\\[\\\\]\\\\(\\\\)\\\\\\\\hello\\\\|\\\\/"`;

exports[`Fetch the reference property of the document. 1`] = `
ArticleModel {
  "content": null,
  "id": null,
  "user": UserModel {
    "id": "AWiYXbY_SjjuUM2b1CGI",
    "name": "enju",
    "version": null,
  },
  "version": null,
}
`;

exports[`Get elasticsearch client. 1`] = `
Object {
  "apiVersion": "5.6",
  "host": "http://localhost:9200",
}
`;
